sirepo/job_driver/__init__.py

```diff
@@ -308,16 +327,30 @@ def _agent_receive_alive(self, msg):
         self._websocket_ready_timeout_cancel()
         if self._websocket:
             if self._websocket != msg.handler:
-                raise AssertionError(pkdformat("incoming msg.content={}", msg.content))
+                self._websocket = None
+                raise AssertionError(
+                    pkdformat("unexpected incoming msg.content={}", msg.content)
+                )
         else:
             self._websocket = msg.handler
         self._websocket_ready.set()
         self._websocket.sr_driver_set(self)
         self._start_idle_timeout()
 
-    def _agent_receive_error(self, msg):
-        # TODO(robnagler) what does this mean? Just a way of logging? Document this.
-        pkdlog("{} msg={}", self, msg)
+    def _agent_receive_job_cmd_stderr(self, msg):
+        """Log stderr from job_cmd"""
+        pkdlog("{} stderr from job_cmd msg={}", self, msg.get("content"))
+
+    def _agent_receive_supervisor(self, msg):
```

Comment:
Can you explain this name? From my reading something like "_agent_receive_unknown_op" is maybe a bit clearer. I think more of a doc would be helpful too. One has to really know our system to know what's going on. I'm struggling to imagine a scenario where we want to act on something without an opId.

Reply:
refactored and changed semantics in 6e0f1ee